var/spack/repos/builtin/packages/ucc/package.py

```diff
@@ -55,5 +60,20 @@ def configure_args(self):
         args = []
         args.extend(self.with_or_without("cuda", activation_value="prefix"))
         args.extend(self.with_or_without("nccl", activation_value="prefix"))
-        # args.extend(self.with_or_without("rccl", activation_value="prefix"))
+        if self.spec.satisfies("+rocm"):
+            rocm_flags = " ".join(
+                [
+                    "-I" + self.spec["hip"].prefix.include,
+                    "-I" + self.spec["hip"].prefix.include.hip,
+                    "-I" + self.spec["hsa-rocr-dev"].prefix.include.hsa,
+                    "-L" + self.spec["hip"].prefix.lib,
+                    "-L" + self.spec["hsa-rocr-dev"].prefix.lib,
+                    "hip_path=" + self.spec["hip"].prefix,
+                ]
+            )
+            args.append("--with-rocm=" + rocm_flags)
+            args.append("--with-ucx=" + self.spec["ucx"].prefix)
```

Comment:
This shouldn't be necessary

Reply:
I thought that as well, but I run into this error if I don't set it:
```
/tmp/root/spack-stage/spack-stage-ucc-1.3.0-ohqojwjegpqp7xtkhfh3kht5oqsu2jya/spack-src/src/utils/ucc_compiler_def.h:12:10: fatal error: 'ucs/type/status.h' file not found
   12 | #include <ucs/type/status.h>
      |          ^~~~~~~~~~~~~~~~~~~
```

It seems that the ucx -I flags are not set if you don't add --with-ucx.

I also found these instructions which say to use --with-ucx when building with ROCm support: https://rocm.docs.amd.com/en/docs-5.0.2/how_to/gpu_aware_mpi.html#collective-operations